### PR TITLE
Fusion: fixing frame range targeting

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Frame range targeting at Rendering instances is now following configured options.

## Testing notes:
1. open fusion workfile and create render instance.
2. test all three options on render target switch. Make sure comp timeline is set to different frame range then asset db and set render range to some different range then comp range so you can test the calculation of handles.
3. ideally test it on multiple rendering targets

Closes https://github.com/ynput/OpenPype/issues/5012

### CODE REVIEW

**Comment by EmberLightVFX** [comment]
> ...and have a validation for when they are outside of the comp's global in/out because Fusion seems to fail rendering for those cases anyway...

I have created a validator that might be of interest to modify and integrate in this PR: https://github.com/ynput/OpenPype/pull/4921

**Comment by BigRoy** [comment]
> > ...and have a validation for when they are outside of the comp's global in/out because Fusion seems to fail rendering for those cases anyway...
> 
> I have created a frame-range validator that might be of interest to modify and integrate in this PR: #4921

Thanks. Took a quick look but unfortunately it's not a perfect fit, but it does show some ideas for improvements to the validator in this PR.

File: openpype/pipeline/publish/abstract_collect_render.py [hunk 1]
```diff
@@ -167,16 +167,27 @@ def process(self, context):
 
             frame_start_render = int(render_instance.frameStart)
             frame_end_render = int(render_instance.frameEnd)
+            """ TODO: Needs to be refofactored because this
+                      seems to be very hacky
+            """
```
Comment (BigRoy) [inline openpype/pipeline/publish/abstract_collect_render.py:172]:
```suggestion
            # TODO: Refactor hacky frame range workaround below
```

**Comment by jakubjezek001** [comment]
We are having bug in the Render Local. It is not restoring original render range after it is rendered.

**Comment by BigRoy** [comment]
@jakubjezek001 could you try again?

**Comment by jakubjezek001** [comment]
> @jakubjezek001 could you try again?

works perfect now and ready to be Approved ;) Thanks @BigRoy

**Review by BigRoy — APPROVED** [review]
I did a lot of the commits so take my approval with a grain of salt please.
@EmberLightVFX anything that still stands out to you?

**Comment by jakubjezek001** [comment]
> I did a lot of the commits so take my approval with a grain of salt please. @EmberLightVFX anything that still stands out to you?

Eh, sorry to be so quick. This was painful at one of our clients and there was a pressure to deploy it quickly. @EmberLightVFX you might want to give some feedback on this please.

**Pull request merged**